### PR TITLE
Last modified header for wide layouts

### LINKED ISSUE #16197: "Modified" file browser column should be "Last Modified" when there is enough space to display
## Description

This affects Notebook 7 downstream: https://github.com/jupyter/notebook/pull/7333#issuecomment-2065867044

In Notebook 7 there is plenty of space to display the full "Last Modified". Also it would be better to keep "Last Modified" in Notebook 7 for parity with the classic notebook, at least for now.

## Reproduce

Available in `4.2.0b2`.

Currently defined here in the code base:

https://github.com/jupyterlab/jupyterlab/blob/c2c8f34a942ec2c7a4fc49887ddce6beb6dc6571/packages/filebrowser/src/listing.ts#L2371

## Expected behavior

The column should say "Last Modified" when the file browser is displayed on the whole page.

There could be the same logic as the column content to reduce it to just "Modified" when the file browser width is smaller.

## Context

Reported in:

- https://github.com/jupyter/notebook/pull/7333#issuecomment-2065867044
- https://github.com/jupyterlab/jupyterlab/pull/15948/files#r1571900809

### PR DESCRIPTION
**Pull request by JasonWeill**
## References

Fixes #16197.

## Code changes

Adds a long "Last Modified" header. The longer header is used when the column is at least 300px wide, suitable for full-screen layouts such as those used in Notebook. The shorter "Modified" header is likely to be used in JupyterLab left panels.

Screenshot 2024-04-19 at 4 56 07 PM

Screenshot 2024-04-19 at 4 56 19 PM

### CODE REVIEW

**Comment by krassowski** [comment]
It looks like this modifies the snapshots. I would guess it is either because of changed thresholds or more likely because of the use of container queries. I would think implementing it by swapping `textContent` would be easier than what is proposed here, but if it works then I think its fine.

**Comment by JasonWeill** [comment]
please update snapshots

**Comment by JasonWeill** [comment]
please update snapshots

**Comment by JasonWeill** [comment]
Rebased and updated — many fewer screenshots this time

File: galata/test/jupyterlab/notebook-run.test.ts-snapshots/notebook-panel-1-jupyterlab-linux.png
Comment (krassowski) [inline galata/test/jupyterlab/notebook-run.test.ts-snapshots/notebook-panel-1-jupyterlab-linux.png:1]: https://github.com/jupyterlab/jupyterlab/issues/15305

File: galata/test/documentation/general.test.ts-snapshots/running-layout-documentation-linux.png
Comment (krassowski) [inline galata/test/documentation/general.test.ts-snapshots/running-layout-documentation-linux.png:1]: https://github.com/jupyterlab/jupyterlab/issues/15103

**Review by krassowski — APPROVED** [review]
Thank you @JasonWeill.

For next iteration: https://github.com/jupyterlab/jupyterlab/issues/16239

**Pull request merged**